From ed9b58042c57f737d5a04ce0e8deba1a97d19e09 Mon Sep 17 00:00:00 2001
From: Andres Salomon <dilinger@queued.net>
Date: Wed, 28 Sep 2022 03:41:16 +0000
Subject: [PATCH] Fix missing cmath header build failure in renderer extension

Building 106.0.5249.51 on a Debian machine, I hit the following
build error:

FAILED: obj/extensions/renderer/renderer/argument_spec.o
clang++ [...] -o obj/extensions/renderer/renderer/argument_spec.o
../../extensions/renderer/bindings/argument_spec.cc:410:16: error: no member named 'isnan' in namespace 'std'
      if (std::isnan(double_val) || std::isinf(double_val)) {
          ~~~~~^
../../extensions/renderer/bindings/argument_spec.cc:410:42: error: no member named 'isinf' in namespace 'std'
      if (std::isnan(double_val) || std::isinf(double_val)) {
                                    ~~~~~^
2 errors generated.

isnan and isinf are both found in the cmath header, so simply
including it fixes that build error.

R=rdevlin.cronin@chromium.org, tjudkins@chromium.org

Change-Id: I95c49a8f90a5a734b62b8988dc3138e3afa3c6f6
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/3913616
Commit-Queue: Andres Salomon <dilinger@queued.net>
Auto-Submit: Andres Salomon <dilinger@queued.net>
Reviewed-by: Tim <tjudkins@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1052196}
---
 extensions/renderer/bindings/argument_spec.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/extensions/renderer/bindings/argument_spec.cc b/extensions/renderer/bindings/argument_spec.cc
index 1e10bf1b561f4..eadd7ec3e3c9d 100644
--- a/extensions/renderer/bindings/argument_spec.cc
+++ b/extensions/renderer/bindings/argument_spec.cc
@@ -4,6 +4,8 @@
 
 #include "extensions/renderer/bindings/argument_spec.h"
 
+#include <cmath>
+
 #include "base/check.h"
 #include "base/strings/string_piece.h"
 #include "base/strings/string_util.h"
